lib/Router.js

```diff
@@ -399,3 +409,32 @@ Router.prototype.middleware = function(options) {
         }.bind(this));
     }.bind(this);
 };
+
+Router.prototype.batch = function(req, res) {
+    var connection = new Connection(this, req, res);
+
+    return Promise.all(req.body.map(function(envelope) {
+        var request = new Request(envelope.url);
+        var parsedUrl = request.getUrl();
+        request.setMethod(envelope.method);
+        request.setResourceId(parsedUrl.pathname);
+
+        if (envelope.body) {
+            request.setResource(envelope.body);
+        }
+
+        return this.handle(request, connection)
+        .then(function(result) {
+            return {
+                status: 200,
+                body: result
+            };
+        })
```

Comment:
@djvirgen why are we hardcoding the http status codes here instead of using the actual result code associated with the result?